pkg/server/api/trustdomain/v1/service.go

```diff
@@ -17,12 +17,14 @@ import (
 
 	trustdomainv1 "github.com/spiffe/spire-api-sdk/proto/spire/api/server/trustdomain/v1"
 	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	bundle_client "github.com/spiffe/spire/pkg/server/bundle/client"
 )
 
 // Config is the service configuration.
 type Config struct {
-	DataStore   datastore.DataStore
-	TrustDomain spiffeid.TrustDomain
+	DataStore     datastore.DataStore
+	TrustDomain   spiffeid.TrustDomain
+	BundleManager *bundle_client.Manager
```

Comment:
Here is the interface I suggested:

```
type BundleRefresher interface {
    // RefreshBundleFor refreshes the bundle for the given trust domain. If the 
    // trust domain is not known, false is returned. Otherwise true is returned along
    // with the result of the refresh operation.
    RefreshBundleFor(ctx context.Context, td spiffeid.TrustDomain) (bool, error)  
}
```

Reply:
Yeah, you're right, @azdagron. I'll define the interface just as in the case of the CA manager. It will facilitate testing.

Thanks o/